webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/MeshFileService.scala

```diff
@@ -372,34 +378,43 @@ class MeshFileService @Inject()(config: DataStoreConfig)(implicit ec: ExecutionC
                     datasetName: String,
                     dataLayerName: String,
                     meshChunkDataRequests: MeshChunkDataRequestList,
-  ): Fox[(Array[Byte], String)] =
+  ): Fox[(Array[Byte], String)] = {
+    val meshFilePath = dataBaseDir
+      .resolve(organizationId)
+      .resolve(datasetName)
+      .resolve(dataLayerName)
+      .resolve(meshesDir)
+      .resolve(s"${meshChunkDataRequests.meshFile}.$hdf5FileExtension")
+    for {
+      resultBox <- meshFileCache.withCachedHdf5(meshFilePath) { cachedMeshFile =>
+        readMeshChunkFromCachedMeshfile(cachedMeshFile, meshChunkDataRequests)
+      }
+      (output, encoding) <- resultBox
+    } yield (output, encoding)
+  }
+
+  private def readMeshChunkFromCachedMeshfile(
+      cachedMeshFile: CachedHdf5File,
+      meshChunkDataRequests: MeshChunkDataRequestList): Box[(Array[Byte], String)] = {
+    val meshFormat = cachedMeshFile.meshFormat
+    // Sort the requests by byte offset to optimize for spinning disk access
+    val requestsReordered =
+      meshChunkDataRequests.requests.zipWithIndex.sortBy(requestAndIndex => requestAndIndex._1.byteOffset).toList
+    val data: List[(Array[Byte], String, Int)] = requestsReordered.map { requestAndIndex =>
+      val meshChunkDataRequest = requestAndIndex._1
+      val data =
+        cachedMeshFile.uint8Reader.readArrayBlockWithOffset("neuroglancer",
+                                                            meshChunkDataRequest.byteSize,
+                                                            meshChunkDataRequest.byteOffset)
+      (data, meshFormat, requestAndIndex._2)
+    }
+    val dataSorted = data.sortBy(d => d._3)
     for {
-      // Sort the requests by byte offset to optimize for spinning disk access
-      data: List[(Array[Byte], String, Int)] <- Fox.serialCombined(
-        meshChunkDataRequests.requests.zipWithIndex.sortBy(requestAndIndex => requestAndIndex._1.byteOffset).toList
-      )(requestAndIndex => {
-        val meshChunkDataRequest = requestAndIndex._1
-        val meshFilePath = dataBaseDir
-          .resolve(organizationId)
-          .resolve(datasetName)
-          .resolve(dataLayerName)
-          .resolve(meshesDir)
-          .resolve(s"${meshChunkDataRequests.meshFile}.$hdf5FileExtension")
-
-        executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
-          val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
-          val data =
-            cachedMeshFile.reader
-              .uint8()
-              .readArrayBlockWithOffset("neuroglancer", meshChunkDataRequest.byteSize, meshChunkDataRequest.byteOffset)
-          (data, meshFormat, requestAndIndex._2)
-        } ?~> "mesh.file.readData.failed"
```

Comment:
Kinda. Since the logic of this function was slightly reordered, this message does not add any more information now that is not already given by the message that is added by the caller.